binary-oracle-pair/program/src/state.rs

```diff
@@ -0,0 +1,93 @@
+//! State transition types
+
+use borsh::{BorshDeserialize, BorshSerialize};
+use solana_program::pubkey::Pubkey;
+use std::mem::size_of;
+
+/// Uninitialized version value, all instances are at least version 1
+pub const UNINITIALIZED_VERSION: u8 = 0;
+/// Initialized pool version
+pub const POOL_VERSION: u8 = 1;
+
+/// Program states.
+#[repr(C)]
```

Comment:
consider addin here https://github.com/solana-labs/solana/blob/193e37018650a0bb4d56246dee6e2892204289bf/frozen-abi/src/abi_digester.rs#L537 .

so each time it is changed, the type system fails and must be fixed (ABI test)